internal/pkg/service/buffer/sink/tablesink/storage/level/staging/config.go

```diff
@@ -7,7 +7,8 @@ import (
 
 // Config contains default configuration for the staging storage.
 type Config struct {
-	Upload UploadConfig `configKey:"upload"`
+	MaxSlicesPerFile int          `configKey:"maxSlicesPerFile" configUsage:"Maximum number of slices in a file, a new file is created after reaching it." validate:"required,min=1,max=50000"`
+	Upload           UploadConfig `configKey:"upload"`
 }
```

Comment:
There are 2 parts of the configuration plus each domains has its own partial configuration.

The first part is `Config` - it is a global configuration of the service.

The second part is `ConfigPatch` - it is modification of the global configuration provided by the user for a specific resource, for example, patch can be set for specific table sink.